src/d3a/models/area/__init__.py

```diff
@@ -193,8 +193,11 @@ def _cycle_markets(self, _trigger_event=True, _market_cycle=False, deactivate=Fa
     def tick(self, is_root_area=False):
         if ConstSettings.IAASettings.MARKET_TYPE == 2 or \
                 ConstSettings.IAASettings.MARKET_TYPE == 3:
-            for market in self._markets.markets.values():
-                market.match_offers_bids()
+            if ConstSettings.GeneralSettings.EVENT_DISPATCHING_VIA_REDIS:
+                self.dispatcher.publish_market_clearing()
```

Comment:
Correct me if Im wrong but this line calls this [method](https://github.com/gridsingularity/d3a/pull/577/files#diff-ad05f547b22979aeada7dea2b3262509R27) which would is only taking care single-sided market.

Reply:
The method you link does not handle one-sided market, only two-sided markets. The first line in this method is a check, that returns immediately if the market is an one-sided market. Generally, one sided markets do not need to clear, therefore this method needs to be called only by the two sided markets.

Reply:
Ah sorry, my bad, you are totally right, I completely overlooked it.